src/com/facebook/buck/io/filesystem/impl/DefaultProjectFilesystem.java

```diff
@@ -967,6 +967,18 @@ public long getPosixFileMode(Path path) throws IOException {
       mode |= MorePosixFilePermissions.toMode(EnumSet.of(PosixFilePermission.OWNER_EXECUTE));
     }
 
+    if (Files.isReadable(path)) {
+      mode |= MorePosixFilePermissions.toMode(EnumSet.of(PosixFilePermission.OWNER_READ));
+      mode |= MorePosixFilePermissions.toMode(EnumSet.of(PosixFilePermission.GROUP_READ));
+      mode |= MorePosixFilePermissions.toMode(EnumSet.of(PosixFilePermission.OTHERS_READ));
+    }
+
+    if (Files.isWritable(path)) {
+      mode |= MorePosixFilePermissions.toMode(EnumSet.of(PosixFilePermission.OWNER_WRITE));
+      mode |= MorePosixFilePermissions.toMode(EnumSet.of(PosixFilePermission.GROUP_WRITE));
+      mode |= MorePosixFilePermissions.toMode(EnumSet.of(PosixFilePermission.OTHERS_WRITE));
+    }
```

Comment:
This fixes the file permission only, but timestamp is still 2/1/1985 12:00AM. Anything we can do about it?